source/more-info/no-url-available.markdown

```diff
@@ -0,0 +1,90 @@
+---
+title: "No URL Available"
+description: "More information on how to resolve the No URL Available message in Home Assistant."
+---
+
+When Home Assistant serves you the "No URL Available" message, you are
+probably trying to set up or configuring an integration that requires your
+account to be linked.
+
+## What happened
+
+The authentication method for linking such an account is called OAuth2, and
+if a modern and more common way of linking systems together, by logging in with
+the provider of the data.
+
+This means Home Assistant will redirect you to the provider to log in. After the
+login was successful, the third-party provider will redirect you back to your
+Home Assistant instance.
+
+For the third-party to be able to redirect/link you back to Home Assistant
+after authentication, Home Assistant needs to tell the third-party application
+about the URL to access Home Assistant on / to redirect back to, unfortunately,
+if this error occurs, Home Assistant hasn't been able to figure out that URL.
+
+## How to solve it
+
+Home Assistant will try to find the URL you are currently using in your browser
+to use as this redirect URL for the authentication. However, Home Assistant
+is only able to determine this, if the URL is known in the configuration.
+
+There are multiple options to consider:
+
+### Using Nabu Casa Home Assistant Cloud
+
+If you have Nabu Casa's Home Assistant Cloud, the easiest way to resolve this,
+is by visiting your Home Assistant instance from the remote UI URL.
+
+If you don't know this URL, surf to <https://remote.nabucasa.com/> and log in.
+You will be automatically redirected to the right address.
+
+Now you can set up the integration as normal, without getting the No URL
+Available message.
+
+### Using your instance by IP address
+
+If you are on your home network, you can visit your instance using the IP
+address of your instance.
+
+For example, `https://192.168.1.2:8123`
+
+Replace the IP with the IP address of your Home Assistant instance. If you
+don't know this, you could also skip this and try the next solution.
+
+Now you can set up the integration as normal, without getting the No URL
+Available message.
+
+### Configuring the instance URL
+
+Another good solution is configuring the Home Assistant instance URL in
+Home Assistant. By letting Home Assistant know about the URL you use to
+access it, Home Assistant will be able to deal with that in s situation
+like this.
+
+Please note, you'll need to enable advanced mode in your user profile in order
+to set this up.
+
+Go to **Configuration** -> **General**.
+
+On this page, two fields that can resolve this issue: "External URL"
+and "Internal URL".
+
+- **Internal URL**: The URL you type in your browser when you are **at home**,
```

Comment:
Should we mention that one should not set this to HTTPS as it will break other things like TTS?

Reply:
Adjusted it using the additional notes below.